modules/unsupported/jdbc-ng/jdbc-hana/src/main/java/org/geotools/data/hana/converter/BooleanConverterFactory.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2018, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.data.hana.converter;
+
+import org.geotools.factory.Hints;
+import org.geotools.util.Converter;
+import org.geotools.util.ConverterFactory;
+
+/**
+ * Converter factory for converting Short objects to Boolean objects.
+ *
+ * <p>In HANA, ResultSet#getObject will return a Short object in case of boolean fields.
+ *
+ * @author Stefan Uhrig, SAP SE
+ */
+public class BooleanConverterFactory implements ConverterFactory {
```

Comment:
This may not cause troubles, but in general, converters are applied everywhere once loaded, e.g., they will be used anywhere else, not just in the HANA code. Right now I don't think you have much of an alternative, doing something else would require changing the core classes (e.g., calling into the dialects for data type conversion). Take this more of as a warning, if something weird happens at runtime while the HANA plugin is running, this might be a cause.

Reply:
Oops, just saw this comment now, my apologies for the similar comment above.